components/server/src/workspace/workspace-starter.ts

```diff
@@ -1424,7 +1424,7 @@ export class WorkspaceStarter {
         }
 
         const volumeSnapshotId =
-            SnapshotContext.is(workspace.context) || WithPrebuild.is(workspace.context)
+            ((SnapshotContext.is(workspace.context) || WithPrebuild.is(workspace.context)) && workspace.context.snapshotBucketId !== undefined && workspace.context.snapshotBucketId != "")
```

Comment:
Hey @sagor999 , here's a brief suggestion to also cover null:

```suggestion
((SnapshotContext.is(workspace.context) || WithPrebuild.is(workspace.context)) && !!workspace.context.snapshotBucketId)
```

[Context](https://www.typescriptlang.org/play?#code/MYewdgziA2CmB0sBOSRIBQEJNgK7WgEoBuAKFEhgWVQ21zABNYAzASzFkZPPCjkQo0WTACJRPCv2pC6YqAFtYAAgAuACw4BzCWSA)

Reply:
[Moar context](https://stackoverflow.com/a/7452731)

Reply:
Thank you!! :heart: